.evergreen/config_generator/components/cse/openssl.py

```diff
@@ -36,16 +37,25 @@
 
 # TODO (CDRIVER-3789): test cse with the 'sharded' topology.
 TEST_MATRIX = [
-    # 4.2 and 4.4 not available on rhel8-zseries.
-    ('rhel8-zseries', 'gcc', None, 'cyrus', ['auth'], ['server'], ['5.0']),
-
-    ('windows-vsCurrent', 'vs2017x64', None, 'cyrus', ['auth'], ['server'], ['4.2', '4.4', '5.0', '6.0' ]),
-
-    # Test 7.0+ with a replica set since Queryable Encryption does not support the 'server' topology. Queryable Encryption tests require 7.0+.
-    ('ubuntu2004',        'gcc',       None, 'cyrus', ['auth'], ['server', 'replica'], ['4.4', '5.0', '6.0', '7.0', '8.0', 'latest']),
-    ('rhel8-zseries',     'gcc',       None, 'cyrus', ['auth'], ['server', 'replica'], [ '7.0', '8.0', 'latest']),
-    ('ubuntu2004-arm64',  'gcc',       None, 'cyrus', ['auth'], ['server', 'replica'], ['4.4', '5.0', '6.0', '7.0', '8.0', 'latest']),
-    ('windows-vsCurrent', 'vs2017x64', None, 'cyrus', ['auth'], ['server', 'replica'], [ '7.0', '8.0', 'latest']),
+    # rhel8-latest provides 4.2+.
+    # Queryable Encryption (6.0+) does not support the single server topology after 7.0+.
```

Comment:
Suggest adding back tests for single on 7.0+.

CSFLE is supported on single+replicaset for 4.2+.
QE is supported in 7.0+. But is not supported on single (requires an oplog).

The `-cse-` tasks run both CSFLE/QE tests. So I think it is worth testing CSFLE on single (the QE tests will just get skipped).

Aside: the naming for In-Use Encryption is inconsistently referenced. I think [Client-Side Encryption > Naming](https://github.com/mongodb/specifications/blob/db693517de561465621b474558ded99eaf38757d/source/client-side-encryption/client-side-encryption.md#naming) is an accurate summary.